drivers/accel/adxl355/adxl355.h

```diff
@@ -0,0 +1,254 @@
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+#ifndef __ADXL355_H__
+#define __ADXL355_H__
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include <stdint.h>
+#include <string.h>
+#include "no-os/i2c.h"
+#include "no-os/spi.h"
+
+/******************************************************************************/
+/******************************** ADXL355 *************************************/
+/******************************************************************************/
+
+/* SPI commands */
+#define ADXL355_SPI_READ         0x01
+#define ADXL355_SPI_WRITE        0x00
+
+
+/* ADXL355 Register Map */
+#define ADXL355_DEVID_AD         0x00
+#define ADXL355_DEVID_MST        0x01
+#define ADXL355_PARTID           0x02
+#define ADXL355_REVID            0x03
+#define ADXL355_STATUS           0x04
+#define ADXL355_FIFO_ENTRIES     0x05
+#define ADXL355_TEMP2            0x06
+#define ADXL355_TEMP1            0x07
+#define ADXL355_XDATA3           0x08
+#define ADXL355_XDATA2           0x09
+#define ADXL355_XDATA1           0x0A
+#define ADXL355_YDATA3           0x0B
+#define ADXL355_YDATA2           0x0C
+#define ADXL355_YDATA1           0x0D
+#define ADXL355_ZDATA3           0x0E
+#define ADXL355_ZDATA2           0x0F
+#define ADXL355_ZDATA1           0x10
+#define ADXL355_FIFO_DATA        0x11
+#define ADXL355_OFFSET_X_H       0x1E
+#define ADXL355_OFFSET_X_L       0x1F
+#define ADXL355_OFFSET_Y_H       0x20
+#define ADXL355_OFFSET_Y_L       0x21
+#define ADXL355_OFFSET_Z_H       0x22
+#define ADXL355_OFFSET_Z_L       0x23
+#define ADXL355_ACT_EN           0x24
+#define ADXL355_ACT_THRESH_H     0x25
+#define ADXL355_ACT_THRESH_L     0x26
+#define ADXL355_ACT_COUNT        0x27
+#define ADXL355_FILTER           0x28
+#define ADXL355_FIFO_SAMPLES     0x29
+#define ADXL355_INT_MAP          0x2A
+#define ADXL355_SYNC             0x2B
+#define ADXL355_RANGE            0x2C
+#define ADXL355_POWER_CTL        0x2D
+#define ADXL355_SELF_TEST        0x2E
+#define ADXL355_RESET            0x2F
+
+
+#define ADXL355_SHADOW_REGISTER_BASE_ADDR  0x50
+#define ADXL355_NB_OF_SHADOW_REGISTERS     0x05
+
+#define ADXL355_DEVID_AD_VAL	0xAD
+#define ADXL355_DEVID_MST_VAL	0x1D
+#define ADXL355_PARTID_VAL	    0xED
+#define ADXL355_REVID_VAL	    0x01
+#define ADXL355_RESET_CODE      0x52
+
+#define ADXL355_XDATA_BYTE_SIZE    0x03
```

Comment:
there is an easier way to do this. Please have a look at the code here:
https://github.com/analogdevicesinc/no-OS/blob/119ba490c1d5762e7092332c3a8abe24e671a922/drivers/adc/ad9250/ad9250.h#L59-L62

Basically, your chip has a fixed number of registers that you know. This number can be coded to fit on 8 bits, or 16 or whatever. In any circumstance, it's generally not bigger than 2^16, that would be a huge number of registers. Since we always use uint32_t for the register address, the top 8 or 16-bits lay unused. So we code some info in the topmost bits regarding the register size in bytes when we define the register address.

This you can use in your adxl355_get_register_value() and adxl355_set_register_value() functions and drop the  adxl355_get_n_consec_reg_val() function.

We have many such examples on no-OS/linux, just search for "_R2B" or "_2B", you'll find them.